File: fm-dddrive/src/main/java/io/dddrive/core/property/model/ReferenceSetProperty.java
package io.dddrive.core.property.model;

import java.util.Set;

import io.dddrive.core.ddd.model.Aggregate;

public interface ReferenceSetProperty<A extends Aggregate> extends Property<A> {

	Set<Object> getItems();

	boolean hasItem(Object aggregateId);

	void clearItems();

	void addItem(Object aggregateId);

	void removeItem(Object aggregateId);

}

File: fm-dddrive/src/main/java/io/dddrive/core/property/model/ReferenceSetProperty.kt
package io.dddrive.core.property.model

import io.dddrive.core.ddd.model.Aggregate

interface ReferenceSetProperty<A : Aggregate> : Property<A> {

	val items: Set<Any>

	fun hasItem(aggregateId: Any): Boolean

	fun clearItems()

	fun addItem(aggregateId: Any)

	fun removeItem(aggregateId: Any)

}
